projects/packages/videopress/src/client/hooks/use-resumable-uploader/index.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+import { useState } from '@wordpress/element';
+import debugFactory from 'debug';
+/**
+ * Internal dependencies
+ */
+import getMediaToken from '../../lib/get-media-token';
+import resumableFileUploader from '../../lib/resumable-file-uploader';
+import { VideoMediaProps } from '../../lib/resumable-file-uploader/types';
+import type React from 'react';
+
+const debug = debugFactory( 'videopress:use-resumable-uploader' );
+
+// eslint-disable-next-line no-shadow
+type UploadingStatusProp = 'idle' | 'resumed' | 'aborted' | 'uploading' | 'done' | 'error';
+
+type UploadingDataProps = {
+	bytesSent: number;
+	bytesTotal: number;
+	percent: number;
+	status: UploadingStatusProp;
+};
+
+type ResumaHandlerProps = {
+	start: () => void;
+	abort: () => void;
+};
+
+type UseResumableUploader = {
+	onUploadHandler: ( event: React.ChangeEvent< HTMLInputElement > ) => void;
+	uploadhandler: ( file: File ) => void;
+	resumeHandler: ResumaHandlerProps;
+	uploadingData: UploadingDataProps;
+	media: VideoMediaProps;
+	error: string;
+};
+
+const useResumableUploader = ( { onProgress, onSuccess, onError } ): UseResumableUploader => {
+	const [ uploadingData, setUploadingData ] = useState< UploadingDataProps >( {
+		bytesSent: 0,
+		bytesTotal: 0,
+		percent: 0,
+		status: 'idle',
+	} );
+
+	const [ media, setMedia ] = useState< VideoMediaProps >();
+	const [ error, setError ] = useState( null );
+	const [ resumeHandler, setResumeHandler ] = useState< ResumaHandlerProps >();
+
+	/**
+	 * Upload a file
+	 *
+	 * @param {File} file - the file to upload
+	 */
+	async function uploadhandler( file: File ) {
```

Comment:
Following the suggestion above, we can use a capital `H`:

```suggestion
	async function uploadHandler( file: File ) {
```